sdk/src/main/java/com/meniga/sdk/models/budget/operators/MenigaBudgetOperationsImp.java

```diff
@@ -55,30 +84,17 @@
     }
 
     @Override
-    public Result<Void> updateBudget(long budgetId, MenigaDecimal targetAmount, DateTime startDate,
-                                     DateTime endDate, List<Long> catIds, int generationTypeValue,
-                                     DateTime repeatUntil) {
-        UpdateBudget req = new UpdateBudget();
-        req.budgetId = budgetId;
-
-        UpdateBudget.UpdateBudgetData data = new UpdateBudget.UpdateBudgetData();
-        data.targetAmount = targetAmount;
-        data.startDate = startDate;
-        data.endDate = endDate;
-        data.categoryIds = catIds;
-        data.generationType = generationTypeValue;
-
-        if (endDate == null/*endDate.getYear() == DateTime.now().getYear() && endDate.getMonthOfYear() == DateTime.now().getMonthOfYear()*/) {
-            data.recurringPattern = null;
-            data.repeatUntil = null;
-        } else {
-            data.recurringPattern = new UpdateBudget.RecurringPattern();
-            data.recurringPattern.monthInterval = Months.monthsBetween(endDate, startDate).getMonths();;
-            data.repeatUntil = endDate;
-        }
-
-        req.rules = new ArrayList<>();
-        req.rules.add(data);
-        return MenigaSDK.executor().updateBudget(req);
+    public Result<Void> updateBudgetRules(long budgetId, CreateBudgetRulesParameters parameters) {
```

Comment:
The name might be confusing to new users, maybe a short explanation re. that updating the rules updates the budget.

Reply:
You're right. I also forgot to deprecate the previous one.